api/response/schema_query.py

```diff
@@ -167,36 +153,48 @@ def search_response_by_room(
         search_result.append(responses)
         return search_result
 
+    def get_all_responses(self, info):
+        response = []
+        rooms = RoomModel.query.all()
+        for room in rooms:
+            room_name = room.name
+            room_response = Response.get_query(info).filter_by(room_id=room.id)
+            response_count = room_response.count()
+            all_responses, total_room_resources = Query.get_room_response(
+                self, room_response, room.id)
+            responses = RoomResponse(
+                room_id=room.id,
+                room_name=room_name,
+                total_responses=response_count,
+                total_room_resources=total_room_resources,
+                response=all_responses)
+            response.append(responses)
+        return response
+
     @Auth.user_roles('Admin')
-    def resolve_all_room_responses(
-            self, info, page=None, per_page=None, lower_limit=None,
-            upper_limit=None, room=None, **kwargs):
-        responses = Query.get_all_reponses(self, info)
-        if isinstance(lower_limit, int) and isinstance(upper_limit, int):
-            responses = Query.filter_rooms_by_responses(
-                self, info, upper_limit, lower_limit
-            )
-        if (
-            (isinstance(lower_limit, int)
-                and not isinstance(upper_limit, int))
-                or (isinstance(upper_limit, int)
-                    and not isinstance(lower_limit, int))):
-            raise GraphQLError(
-                "Provide upper and lower limits to filter by response number")
-        if (
-            isinstance(lower_limit, int)
-                and isinstance(upper_limit, int)
-                and room):
-            responses = Query.search_response_by_room(
-                self, info, upper_limit, lower_limit, room)
-        if not upper_limit and not lower_limit and room:
+    def resolve_all_room_responses(self, info, **kwargs):
+        responses = Query.get_all_responses(self, info)
+        check_limits_are_provided(
+            kwargs.get('lower_limit_count'),
+            kwargs.get('upper_limit_count'), int
+        )
+        responses = filter_responses(
+            int, filter_rooms_by_responses, kwargs.get('upper_limit_count'),
+            kwargs.get('lower_limit_count'),  Query, info, responses
+        )
+        responses = check_response_and_room(
+            int, info, kwargs.get('room'), kwargs.get('upper_limit_count'),
+            kwargs.get('lower_limit_count'), responses, Query
+        )
+        if not kwargs.get('upper_limit_count') and not kwargs.get('lower_limit_count') and kwargs.get('room'): # noqa
```

Comment:
I believe you don't need this `noqa`

Reply:
Also `('upper_limit_count' and 'lower_limit_count' not in kwargs)` would apply

Reply:
Thank you, this has been implemented